src/help.cpp

```diff
@@ -18,17 +18,23 @@ Help::Help(QObject *parent): QObject(parent),texDocSystem(0)
  */
 void Help::execTexdocDialog(const QStringList &packages, const QString &defaultPackage)
 {
-    TexdocDialog dialog(nullptr,this);
+	TexdocDialog dialog(nullptr,this);
 	dialog.setPackageNames(packages);
+	if (packages.count()==0) {
+		mShowAllPackages = true;
+	}
+	dialog.setShowAllPackages(mShowAllPackages);
 	if (!defaultPackage.isEmpty()) {
 		dialog.setPreferredPackage(defaultPackage);
-    }else{
-        dialog.setShowAllPackages(mShowAllPackages);
-    }
+	}
 	if (dialog.exec()) {
 		viewTexdoc(dialog.selectedPackage());
 	}
-    mShowAllPackages=dialog.showAllPackages();
+	if (packages.count()==0) {
+		mShowAllPackages = false;

```

Comment:
It should not force mShowAllPackages to false either. It just should not change the previous state which maybe true or false.

Reply:
@sunderme build is running